source/common/stream_info/stream_info_impl.h

```diff
@@ -213,6 +203,13 @@ struct StreamInfoImpl : public StreamInfo {
     return downstream_remote_address_;
   }
 
+  void
+  setDownstreamAddresses(const Network::ConnectedSocketAddressProvider& address_provider) override {
+    downstream_local_address_ = address_provider.localAddress();
```

Comment:
Alternate idea: could this function just store a reference to the `address_provider` and call methods on it as the implementation for getting these addresses? That way, the addresses aren't latched here, so if they're updated later on the connection/request/whatever, it will be automatically reflected in the streaminfo.

Reply:
I thought about that, but I would be concerned about lifetime issues as stream info can stick around for quite some time. I think if we did that it would have to be a shared_ptr, and then probably an object shared between the socket and here? WDYT?

Reply:
Yeah, lifetime could be tricky. This is probably good enough for now; we could try to make it better in the future.